From b26bdc8987ee4149a35a6c5bd3d2b4ae41270e6a Mon Sep 17 00:00:00 2001
From: Firestorm_6.3.3_VR_Source
Date: Mon, 4 Mar 2024 19:11:32 -0500
Subject: [PATCH] P373R 6.6.8 baseline diff

---
 llviewerdisplay.cpp | 44 ++++++++++++++++++++++++++++++++++++++++++--
 1 file changed, 42 insertions(+), 2 deletions(-)

diff --git a/llviewerdisplay.cpp b/llviewerdisplay.cpp
index d46fbad..9d643aa 100644
--- a/llviewerdisplay.cpp
+++ b/llviewerdisplay.cpp
@@ -87,6 +87,11 @@
 #include "fsdata.h"
 #include "fsperfstats.h" // <FS:Beq> performance stats support
 
+//################################### P373R ######################################
+#include "llviewerVR.cpp"
+llviewerVR gVR;
+//################################### END P373R ##################################
+
 extern LLPointer<LLViewerTexture> gStartTexture;
 extern bool gShiftFrame;
 
@@ -783,6 +788,11 @@ void display(BOOL rebuild, F32 zoom_factor, int subfield, BOOL for_snapshot)
 			gPipeline.toggleRenderType(LLPipeline::RENDER_TYPE_HUD_PARTICLES);
 		}
 
+		//################################### P373R ######################################
+		sec:
+		gVR.ProcessVRCamera();
+		//################################### END P373R ##################################
+		
 		stop_glerror();
 		display_update_camera();
 		stop_glerror();
@@ -1158,7 +1168,29 @@ void display(BOOL rebuild, F32 zoom_factor, int subfield, BOOL for_snapshot)
 		if (!for_snapshot)
 		{
 			render_ui();
-			swap();
+			//################################### P373R ######################################
+			gVR.vrDisplay();
+
+			//################################### END P373R ######################################
+			//swap();
+			//################################### P373R ######################################
+			if (gVR.leftEyeDesc.IsReady  && !gVR.rightEyeDesc.IsReady && gVR.m_fEyeDistance > 0)
+			{
+				goto sec;
+
+
+			}
+			if (!gVR.leftEyeDesc.IsReady  && !gVR.rightEyeDesc.IsReady)
+			{
+				//gVR.HandleInput();
+
+			}
+			if (!gVR.m_bVrActive)
+				swap();
+
+			//################################### END P373R ##################################
+			//swap();
+			
 		}
 
 		
@@ -1593,6 +1625,9 @@ void render_ui_3d()
 	
 	gUIProgram.bind();
 
+	//################################### P373R ######################################
+	gVR.RenderControllerAxes();
+	//################################### END P373R ##################################
 	// Coordinate axes
 	// <FS:Ansariel> gSavedSettings replacement
 	//if (gSavedSettings.getBOOL("ShowAxes"))
@@ -1731,7 +1766,9 @@ void render_ui_2d()
 	{
 		gViewerWindow->draw();
 	}
-
+	//################################### P373R ######################################
+	gVR.DrawCursors();
+	//################################### END P373R ##################################
 
 
 	// reset current origin for font rendering, in case of tiling render
@@ -1817,6 +1854,9 @@ void render_disconnected_background()
 
 void display_cleanup()
 {
+	//################################### P373R ######################################
+		gVR.vrStartup(TRUE);
+	//################################### END P373R ######################################
 	gDisconnectedImagep = NULL;
 }
 
-- 
2.40.1

